source/common/http/http2/codec_impl.cc

```diff
@@ -344,6 +349,20 @@ void ConnectionImpl::StreamImpl::saveHeader(HeaderString&& name, HeaderString&&
 }
 
 void ConnectionImpl::StreamImpl::submitTrailers(const HeaderMap& trailers) {
+  // Some browsers (e.g. WebKit-based browsers: https://bugs.webkit.org/show_bug.cgi?id=210108) have
+  // a problem with processing empty trailers (END_STREAM | END_HEADERS with zero length HEADERS) of
+  // an HTTP/2 response as reported here: https://github.com/envoyproxy/envoy/issues/10514.
+  const bool skip_encoding_empty_trailers =
+      Runtime::runtimeFeatureEnabled("envoy.reloadable_features.skip_encoding_empty_trailers");
```

Comment:
per other comment, this can be latched, also per my other comment I don't think we need to change the other function.